src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -199,85 +200,80 @@
 #   UNNEST(), as BigQuery. DB2 is not currently supported by SQLFluff.
 ansi_dialect.sets("value_table_functions").update([])
 
+
+class IdentifierSegment(CodeSegment):
+    """An identifier segment.
+
+    Defined here for type inheritance.
+    """
+
+    type = "identifier"
+
+
+class LiteralSegment(CodeSegment):
+    """An literal segment.
+
+    Defined here for type inheritance.
+    """
+
+    type = "literal"
+
+
+class LiteralKeywordSegment(KeywordSegment):
+    """An keyword style literal segment.
+
+    Defined here for type inheritance.
+    """
+
+    type = "literal"
+
+
 ansi_dialect.add(
     # Real segments
     DelimiterGrammar=Ref("SemicolonSegment"),
-    SemicolonSegment=StringParser(
-        ";", SymbolSegment, name="semicolon", type="statement_terminator"
-    ),
-    ColonSegment=StringParser(":", SymbolSegment, name="colon", type="colon"),
-    SliceSegment=StringParser(":", SymbolSegment, name="slice", type="slice"),
-    StartBracketSegment=StringParser(
-        "(", SymbolSegment, name="start_bracket", type="start_bracket"
-    ),
-    EndBracketSegment=StringParser(
-        ")", SymbolSegment, name="end_bracket", type="end_bracket"
-    ),
+    SemicolonSegment=StringParser(";", SymbolSegment, type="statement_terminator"),
+    ColonSegment=StringParser(":", SymbolSegment, type="colon"),
+    SliceSegment=StringParser(":", SymbolSegment, type="slice"),
+    StartBracketSegment=StringParser("(", SymbolSegment, type="start_bracket"),
+    EndBracketSegment=StringParser(")", SymbolSegment, type="end_bracket"),
     StartSquareBracketSegment=StringParser(
-        "[", SymbolSegment, name="start_square_bracket", type="start_square_bracket"
-    ),
-    EndSquareBracketSegment=StringParser(
-        "]", SymbolSegment, name="end_square_bracket", type="end_square_bracket"
+        "[", SymbolSegment, type="start_square_bracket"
     ),
+    EndSquareBracketSegment=StringParser("]", SymbolSegment, type="end_square_bracket"),
     StartCurlyBracketSegment=StringParser(
-        "{", SymbolSegment, name="start_curly_bracket", type="start_curly_bracket"
```

Comment:
For the changes in this section, it looks like we're losing whatever `name` we were setting previously. Is that a concern?

Reply:
I was thinking about this. Where the answer is _yes_ then I'll create another parent segment (like what I've done with identifiers) - where the answer is _no_ then I'll just leave it.